src/main/java/org/openstreetmap/atlas/checks/validation/linear/edges/SuddenHighwayTypeChangeCheck.java

```diff
@@ -0,0 +1,248 @@
+
+        // Check ways' first edge's connected edges for suspiciously large highway tag jumps.
+        for (final Edge firstEdgeEdge : firstEdgeStartNodeEdges)
+        {
+            final Optional<HighwayTag> firstEdgeEdgeHighwayTagOptional = HighwayTag
+                    .highwayTag(firstEdgeEdge);
+            if (firstEdgeEdgeHighwayTagOptional.isPresent()
+                    && edgeBeingVerifiedHighwayTagOptional.isPresent()
+                    && !firstEdgeStartNodeEdgesHighwayTags.contains(edgeBeingVerifiedHighwayTag)
+                    && !this.edgeIsRoundaboutOrCircular(firstEdgeEdge))
+            {
+                final HighwayTag firstEdgeEdgeHighwayTag = firstEdgeEdgeHighwayTagOptional.get();
+
+                // All cases
+                if (this.isCaseOne(edgeBeingVerifiedHighwayTag, firstEdgeEdgeHighwayTag)
+                        || this.isCaseTwo(edgeBeingVerifiedHighwayTag, firstEdgeEdgeHighwayTag)
+                        || this.isCaseThree(edgeBeingVerifiedHighwayTag, firstEdgeEdgeHighwayTag))
+                {
+                    markAsFlagged(edgeBeingVerified.getOsmIdentifier());
+                    return Optional.of(this.createFlag(object,
+                            this.getLocalizedInstruction(0, object.getOsmIdentifier())));
+                }
+            }
+        }
+
+        // Check ways' last edge's connected edges for suspiciously large highway tag jumps
+        for (final Edge lastEdgeEdge : lastEdgeEndNodeEdges)
+        {
+            final Optional<HighwayTag> lastEdgeEdgeHighwayTagOptional = HighwayTag
+                    .highwayTag(lastEdgeEdge);
+            if (lastEdgeEdgeHighwayTagOptional.isPresent()
+                    && edgeBeingVerifiedHighwayTagOptional.isPresent()
+                    && !lastEdgeEndNodeEdgesHighwayTags.contains(edgeBeingVerifiedHighwayTag)
+                    && !this.edgeIsRoundaboutOrCircular(lastEdgeEdge))
+            {
+                final HighwayTag lastEdgeEdgeHighwayTag = lastEdgeEdgeHighwayTagOptional.get();
+
+                // All cases
+                if (this.isCaseOne(edgeBeingVerifiedHighwayTag, lastEdgeEdgeHighwayTag)
+                        || this.isCaseTwo(edgeBeingVerifiedHighwayTag, lastEdgeEdgeHighwayTag)
+                        || this.isCaseThree(edgeBeingVerifiedHighwayTag, lastEdgeEdgeHighwayTag))
+                {
+                    markAsFlagged(edgeBeingVerified.getOsmIdentifier());
+                    return Optional.of(this.createFlag(object,
+                            this.getLocalizedInstruction(0, object.getOsmIdentifier())));
+                }
+            }
+        }
+        markAsFlagged(edgeBeingVerified.getOsmIdentifier());
+        return Optional.empty();
+    }
+
+    @Override
+    protected List<String> getFallbackInstructions()
+    {
+        return FALLBACK_INSTRUCTIONS;
+    }
+
+    private boolean edgeBeingVerifiedCaseOne(final HighwayTag edgeHighwayTag)
```

Comment:
Hate to be a stickler, but some method docs would be helpful, if not actually the standard for new checks

Reply:
updated!